### PR TITLE
[Merged by Bors] - feat(Data/List/Chain): `chain'_attachWith`

### PR DESCRIPTION
**Pull request by vihdzp**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/List/Chain.lean [hunk 1]
```diff
@@ -319,6 +319,31 @@ lemma chain'_join : ∀ {L : List (List α)}, [] ∉ L →
...
+    intro hc b (hb : _ = _)
+    · simp_rw [hb, Option.pbind_some] at hc
+      have hb' := h b (mem_cons_of_mem a (mem_of_mem_head? hb))
+      exact ⟨h a (mem_cons_self a l), hb', hc ⟨b, hb'⟩ rfl⟩
+    · cases l with
+      | nil => cases hb
+      | cons b l =>
+        obtain ⟨_, _, rfl⟩ := hb
+        obtain ⟨_, _, hr⟩ := hc b rfl
+        exact hr
```
Comment (YaelDillies) [inline Mathlib/Data/List/Chain.lean:341]: Can aesop golf any of this?

Comment (vihdzp) [reply Mathlib/Data/List/Chain.lean]: Yes! The whole `cases l` block can be replaced by `cases l <;> aesop`.

Comment (YaelDillies) [reply Mathlib/Data/List/Chain.lean]: Ahah, my guess was that aesop would do the other case

**Comment by YaelDillies** [comment]
Thanks!

maintainer merge

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**